docs/diga_api.md

```diff
@@ -0,0 +1,289 @@
+
+- prescription code
+- DiGA id of the manufacturer
+- time period for prescription
+- information about the invoice issuer and the insurer
+
+Additional information can be found on this [wiki page](https://github.com/alex-therapeutics/diga-api-client/wiki/Billing-validation---XRechnung).
+
+## Encryption
+
+Given that we are dealing with patient data, it is required to properly encrypt the data sent within requests.
+The encryption is based on certificates which need to be requested at the ITSG which acts as a trust center.
+You can request it at [ITSG website](https://www.itsg.de/produkte/trust-center/zertifikat-beantragen/), assuming that you already have an IK number.
+
+A technical documentation for the encryption (of course in German) can be found on the [technical standards page at gkv](https://www.gkv-datenaustausch.de/technische_standards_1/technische_standards.jsp) as [attachment 16](https://www.gkv-datenaustausch.de/media/dokumente/standards_und_normen/technische_spezifikationen/Anlage_16.pdf) under `Security Schnittstelle (SECON)` section.
+Once the certificates are issued, checkout the [prerequisites from the Readme](https://github.com/alex-therapeutics/diga-api-client#prerequisites) to be able to use them with the api client.
+
+## Request processing time
+
+According to the documentation, the processing time constraints are as follows:
+
+Mean value 10000[msec], 95% quantile [msec] 15000
+
+Is that really 10 seconds for a single request?
+
+# FAQ
+
+## What can I do if I want to make requests but my DiGA is not approved yet and I therefore do not have a diga id?
+
+Just sent some 5 digit number as part of the `nutzdaten`, e.g. `12345`.
+
+## What can I do if I dont know the IK number yet and still want to test the client?
+
+You will need an IK number to request the certificate for encryption and decryption. Therefore, it is a must-have.
+
+## Are there solutions for non-java users?
+
+So far, there are no other clients available.
+A solution could be to create a docker image which exposes the java client via an api.
+Additionally, we hope that this documentation provides enough information so that clients in other languages can be developed.
+
+## My DiGA (app) is only valid for 30 days. Can users request a fresh code from their insurer while their existing code is still valid?
+
+**TODO** Hopefully, users want to continue using your DiGA after their initial prescription code expired.
+
+- Can they already request a new code while still being active?
+- How will the verification and billing api handle these cases?
+- Is the `TagDerLeistungserbringung` the date on which the request is made or is this set by the insurer when issuing the prescription code?
+
+## Does the diga-api-client expect prescription codes in a special format, e.g. all capitalized?
+
+The insurance information are extracted based on the prescription code from the mapping file. Therefore, the code must be capitalized.
+
+**TODO**: How do the apis handle this?
```

Comment:
According to [this document](https://www.gkv-datenaustausch.de/media/dokumente/leistungserbringer_1/digitale_gesundheitsanwendungen/technische_anlagen_aktuell_7/DiGA_Anlage_1_Technische_Anlage_zur_RL_V1.2_20210525.pdf) on page 9, the characters can be of `[A-Z and 2-7]`

Therefore, I assume codes should be send in capitalized form.

Reply:
Will do as soon as we have all the prerequisites :)
I guess for the mapping file it could cause troubles when the code is entered in lower-case. But that could probably be fixed in this library. Generally, it could make sense to capitalize the code in this library?

Reply:
Removed this question and extended the section on the code structure. given that the code regex is described in the document, I really assume the code must be capitalized.